bootloader/src/pyi_utils.c

```diff
@@ -261,6 +261,50 @@ pyi_unsetenv(const char *variable)
 
 #ifdef _WIN32
 
+/* Resolve the runtime tmpdir path and build nested directories */
+wchar_t 
+*pyi_build_temp_folder(char *runtime_tmpdir)
+{
+    wchar_t *wruntime_tmpdir;
+    wchar_t wruntime_tmpdir_expanded[PATH_MAX];
+    wchar_t *wruntime_tmpdir_abspath;
+    wchar_t *cursor;
+    wchar_t path_builder[PATH_MAX];
+    DWORD rc;
+    // Expand environment variables like %LOCALAPPDATA%
+    wruntime_tmpdir = pyi_win32_utils_from_utf8(NULL, runtime_tmpdir, 0);
+    if (!wruntime_tmpdir) {
+        FATALERROR("LOADER: Failed to convert runtime-tmpdir to a wide string.\n");
+        return NULL;
+    }
+    rc = ExpandEnvironmentStringsW(wruntime_tmpdir, wruntime_tmpdir_expanded, PATH_MAX);
+    free(wruntime_tmpdir);
+    if (!rc) {
+        FATALERROR("LOADER: Failed to expand environment variables in the runtime-tmpdir.\n");
+        return NULL;
+    }
+    // Get the absolute path
+    wruntime_tmpdir_abspath = _wfullpath(NULL, wruntime_tmpdir_expanded, PATH_MAX);
+    if (!wruntime_tmpdir_abspath) {
+        FATALERROR("LOADER: Failed to obtain the absolute path of the runtime-tmpdir.\n");
+        return NULL;
+    }
+    VS("LOADER: absolute runtime tmpdir is %ls\n", wruntime_tmpdir_abspath);
+    // Create the directory path if it does not yet already exist (e.g. %AppData%\NewFolder\NestedFolder)
+    ZeroMemory(path_builder, PATH_MAX * sizeof(wchar_t));
+    cursor = wcschr(wruntime_tmpdir_abspath, L'\\');
+    while(cursor != NULL) {
+        wcsncpy(path_builder, wruntime_tmpdir_abspath, cursor - wruntime_tmpdir_abspath + 1);
+        CreateDirectoryW(path_builder, NULL);
```

Comment:
There's no error handling here. But I suppose if we failed to create a directory here, we will also fail to create the actual _MEIPASS dir inside the runtime_tmpdir later on, so the error will be effectively caught there. So aside from not stopping early, this is probably not an issue?